examples/complex_expr.py

```diff
@@ -0,0 +1,53 @@
+"""This example showcases how to string together complex expressions.
```

Comment:
Any chance you can make this an example in [here](https://github.com/argoproj-labs/hera-workflows/tree/main/docs/examples) and add it to [this list](https://github.com/argoproj-labs/hera-workflows/blob/main/docs/index.rst)? I think it would complement the documentation written for expr